rust_programs/ttf_renderer/src/glyphs.rs

```diff
@@ -562,3 +573,107 @@ fn interpret_values_via_flags(
     }
     values
 }
+
+fn bezier_2d(start: isize, ctrl: isize, end: isize, t: f64) -> isize {
+    return ((start as f64) + (2 as f64)*t*(ctrl as f64) + t*t*((end - ctrl) as f64)).round() as isize;
+}
+
+fn points_for_polygon(
+    start_index: usize,
+    end_index: usize,
+    points: &Vec<isize>,
+    flags: &Vec<Vec<GlyphOutlineFlag>>,
+    interpolate: bool
+) -> Vec<isize> {
+    
+    let mut values = vec![];
+
+    if !interpolate {
+        return points[start_index..=end_index].to_vec().clone();
+    }
+    
+    /*
+    for index in start_index..=end_index {
+        values.push(points[index]);
+    }
+    return values;
+    */
+
+    if start_index == end_index {
```

Comment:
Seems like this shouldn't happen? It'd imply that the font glyph specified an empty contour

Reply:
Since we can't guarantee that this is never the case, I prefer having this safety measure rather than forcing the resonsibility on the caller. Having empty countours does not seem out of the question to me, in terms of how container formats are often used.